message-handler.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Message Handler - Manages communication with parent window
+ * Handles postMessage API communication for iframe integration
+ */
+export class MessageHandler {
```

Comment:
I really like how you did messages here and we should perhaps look at it for all the other iframe tools, although we cannot control them.